sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -249,16 +330,74 @@
   }
 
   /**
+   * Creates an uninitialized {@link Read} {@link PTransform}, using a Kafka {@link Deserializer}
+   * based on {@link AvroCoder}. The coder reads Avro data directly without using an Avro object
+   * container.
+   */
+  @SuppressWarnings("unchecked")
+  public static <K, V> Read<K, V> fromAvro(Class<K> keyClass, Class<V> valueClass) {
+    // Kafka constructs deserializers directly. Pass coder through consumer
+    // configuration.
+    ImmutableMap.Builder<String, Object> builder = new ImmutableMap.Builder<>();
+    Map<String, Object> config = builder
+            .putAll(Read.DEFAULT_CONSUMER_PROPERTIES)
```

Comment:
This would serve as an example for others to their own coders as serializers.
They wouldn't have access to `Read.DEFAULT_CONSUMER_PROPERTIES` and other builders. Can you rewrite it using only the public api?

In fact you could implement `fromCoders()` (again using only public API) and fromAvro would just invoke that.